### DISCUSSION TITLE
Use tls/ssl in fo-dicom Client/Server communication

### DISCUSSION THREAD

**Discussion by Ealfaro41** [fo-dicom/fo-dicom#1558] [Q&A]
Greetings fo-dicom community, I have a client (SCU) and a server (SCP) deployed with fo.dicom 4.0.7. The client sends images to the server over the internet. How can I configure the use of tls/ssl for transmission? 
I have tried to enable the use of tls in the creation of the client by means of: 

[Dicom.Network.Client.DicomClient("IP", "Port", true, "AETitle", "AETitle, 5000, 10000, 2000);] 

but it gives error [SSL client authentication took longer than 60s].

Any idea what to enable on the server? or how to implement tls/ssl? Any idea or information is useful since I don't know where to start.

Greetings

**Comment by amoerie** [comment]
The latest version of fo-dicom (5.x) has support for TLS. The 4.x release train will only have urgent security patches in the future, nothing else. We are no longer actively developing on the 4.x version.

In the 5.x version, you can pass in an implementation of ITlsInitiator when sending requests, and an instance of ITlsAcceptor when receiving them.
Review the appropriate unit tests for code samples here: https://github.com/fo-dicom/fo-dicom/blob/development/Tests/FO-DICOM.Tests/Network/Client/DicomClientTlsTest.cs
